src/internal/connector/onedrive/item.go

```diff
@@ -66,35 +69,17 @@ func oneDriveItemReader(
 // and using a http client to initialize a reader
 func driveItemReader(
 	ctx context.Context,
-	service graph.Servicer,
-	driveID, itemID string,
-) (models.DriveItemable, io.ReadCloser, error) {
-	logger.Ctx(ctx).Debugw("Reading Item", "id", itemID, "time", time.Now())
-
-	item, err := service.Client().DrivesById(driveID).ItemsById(itemID).Get(ctx, nil)
-	if err != nil {
-		return nil, nil, errors.Wrapf(err, "failed to get item %s", itemID)
-	}
-
-	// Get the download URL - https://docs.microsoft.com/en-us/graph/api/driveitem-get-content
-	// These URLs are pre-authenticated and can be used to download the data using the standard
-	// http client
-	if _, found := item.GetAdditionalData()[downloadURLKey]; !found {
-		return nil, nil, errors.Errorf("file does not have a download URL. ID: %s, %#v",
-			itemID, item.GetAdditionalData())
-	}
-
-	downloadURL := item.GetAdditionalData()[downloadURLKey].(*string)
-
+	url string,
```

Comment:
Pulling @ashmrtn into this PR as well, for additional perspective on restartability.  I imagine this will handle it fine as is, since were just grabbing a super-set of the id alone.  But wanted to double-check.

Reply:
yea, it should be fine when we start adding restartability later on. The extra data will just need to be serialized in the ID file along with the rest. We'll need to make sure we make a service-dependent way to serialize/deserialize all that